### ISSUE TITLE
Upload file receiver implementations broken

### ISSUE THREAD

**Issue by chemmett** [vaadin/flow#4860]
The file-based Receiver implementations (FileBuffer and MultiFileBuffer) appear to be broken because once the upload is complete, getFileDescriptor() returns an invalid FileDescriptor, so getInputStream() doesn't work either (returns an input stream that's already been closed.) The only time getFileDescriptor() is valid is while the upload is in progress.

Unfortunately this makes these implementations nearly useless because you can't get an input stream to the file after the upload is complete, nor can you get the temporary file name on disk to open it yourself. They "work" in that they allow you to store uploaded files in the system temporary directory, but then you have no way of doing anything with them.

    FileBuffer buffer = new FileBuffer();
    Upload upload = new Upload(buffer);
    upload.addSucceededListener(event -> {
        // buffer.getFileDescriptor().valid() == false
        // buffer.getInputStream() returns a closed stream
    });

Expected: getInputStream() should return a valid stream after the upload has succeeded.

Actual: getInputStream() returns a closed stream after the upload has succeeded.

Vaadin 10.0.9
Java 8
Tested on OSX and Linux

**Comment by denis-anisimov** [comment]
This bug is in the wrong repo. It should be filed against `vaadin-upload-flow`.

But I just close the bug since it's a duplicate of https://github.com/vaadin/vaadin-upload-flow/issues/105
